modules/mongodb/mongodb.go

```diff
@@ -122,16 +105,66 @@ func (c *MongoDBContainer) ConnectionString(ctx context.Context) (string, error)
 	if err != nil {
 		return "", err
 	}
+
+	connStr := strings.Builder{}
+	connStr.WriteString("mongodb://")
+
 	if c.username != "" && c.password != "" {
+		connStr.WriteString(fmt.Sprintf("%s:%s@", c.username, c.password))
 		return fmt.Sprintf("mongodb://%s:%s@%s:%s", c.username, c.password, host, port.Port()), nil
```

Comment:
IIUC from this block, if there is an username and password, the connection string won't have anything about the replicaSet, even though it was passed as an option, right?

If not sure, you can extract the url-builder part to a function and add unit tests for it